app/src/main/java/com/jerboa/feat/ModActions.kt

```diff
@@ -45,15 +45,30 @@ fun futureDaysToUnixTime(days: Long?): Long? {
 }
 
 fun amMod(
-    moderators: List<CommunityModeratorView>?,
-    myId: PersonId?,
+    moderators: List<PersonId>?,
+    myId: PersonId,
 ): Boolean {
-    return moderators?.map { it.moderator.id }?.contains(myId) ?: false
+    return moderators?.contains(myId) ?: false
 }
 
-fun amAdmin(
-    admins: List<PersonView>?,
-    myId: PersonId?,
-): Boolean {
-    return admins?.map { it.person.id }?.contains(myId) ?: false
+/**
+ * In screens with posts from different communities we don't have access to moderators of those communities
+ * So that means that non admin mods can't moderate those posts from that screen
+ *
+ * So this is QoL were we simulate the mods of the community
+ * It is not completely accurate as it doesn't take into account the hierarchy of mods
+ */
```

Comment:
This really shows that the `canMod` logic needs to be added on the back-end: https://github.com/LemmyNet/lemmy/issues/4365

Once I eventually get some time to work on that, we'd be able to remove the mod list from almost every response, and code like this.

Reply:
One thing I forgot, how does the `self` option play into this? Which mod actions can you do on yourself? like lockpost? How would that work with canMod from server side?

Currently we don't do that correctly either. canMod should probably return which actions can be done?

Reply:
I just looked through the lemmy-ui code, and its only `distinguish` that you can do to yourself, and anyone lower than you. We could probably change that to just `isModOrAdmin && myComment`

Reply:
Only distinguish? Can top admin not pin his own comment? Or lock his own posts?